### PR TITLE
Topics/waloo/doc webhelp

### PR DESCRIPTION
**Pull request by kolekti**
PR for comments

### CODE REVIEW

**Comment by amake** [comment]
Thank you! I appreciate it.

**Comment by amake** [comment]
The check failure is due to an unrelated issue, so don't worry about it.

**Review by amake — COMMENTED** [review]
I pushed some whitespace fixups and rebased on top of the latest master.

For some reason GitHub thinks that the latest master commit is also part of your PR; please ignore it.

File: doc_src/build.xml [hunk 1]
```diff
@@ -178,12 +218,99 @@
...
+             <arg value="${language}/xhtml5/header.xhtml" />
+             <arg value="${language}/index.xml" />
+             <arg value="whc-header.xsl" />
+             <arg value="language=${language}"/>
+             <classpath>
+               <pathelement location="${saxon}" />
+               <pathelement path="${saxon}" />
+             </classpath>
+        </java>      
+        <copy file="omegat.svg" tofile="../docs/${language}/images/omegat.svg"/>
```
[the remark this answers is not in the document]
Comment (kolekti) [reply doc_src/build.xml:284]:
```suggestion
        <copy file="../images/OmegaT.svg" tofile="../docs/${language}/images/omegat.svg"/>
```

[the remark this answers is not in the document]
Comment (kolekti) [reply doc_src/build.xml:284]: > We have an existing logo SVG at [images/OmegaT.svg](https://github.com/omegat-org/omegat/blob/master/images/OmegaT.svg); can we not use that rather than add a new logo?

Yes, definitly, I missed it. So you can remove the doc_src/omegat.svg from git

File: doc_src/build.xml [hunk 2]
```diff
@@ -178,12 +218,99 @@
...
+    
+    <target name="pictures">
         <copy todir="../docs/${language}/images">
              <fileset dir="${language}/images" />
         </copy>
     </target>
 
-    <target name="html" depends="chunk,instant-start">
+    <target name="html" depends="pictures,chunk,instant-start">
+    </target>
```
[the remark this answers is not in the document]
Comment (kolekti) [reply doc_src/build.xml]: The instant start page is not generated by the html5 task, I add the task to the "all" rule and remove html task.
That's also why every OmegaT.css stylesheet shiould remain in doc_src/$language directories.

File: doc_src/en/messages.xml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<entries>
+  <entry id="appendices">Appendices</entry>
```
[the remark this answers is not in the document]
Comment (kolekti) [reply doc_src/en/messages.xml]: Right, this file contains labels to be included in the help, that are not explicit in doc sources neither in whc labels.

Note that some labels are included by whc which has its own localization. Whc localization does not include as much languages than OmageT help, this could cause an issue in the future.

File: doc_src/build.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (amake) [reply doc_src/build.xml]: Thanks!

File: doc_src/build.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (amake) [reply doc_src/build.xml]: OK that's helpful.

The way I update the manual materials currently is through Gradle. I have three tasks:

- `manualHtmls` runs `docgen chunk`
- `manualPdfs` runs `docgen pdf`
- `instantStartGuides` runs `docgen instant-start`

So in other words I intentionally split `html` apart anyway.

It sounds like now we can have:

- `manualHtmls` runs `docgen html5`
- `manualPdfs` runs `docgen pdf`
- `instantStartGuides` runs `docgen instant-start`
- Delete the `html` and `chunk` targets

Does that sound right?

File: doc_src/en/messages.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (amake) [reply doc_src/en/messages.xml]: OK, thanks.

File: doc_src/build.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (kolekti) [reply doc_src/build.xml]: That's ok

File: doc_src/build.xml [hunk 4]
```diff
@@ -120,36 +120,76 @@
...
+      <java dir="."
+            fork="true"
+            failonerror="true"
+            maxmemory="128m"
+            classname="com.icl.saxon.StyleSheet" >
+        <arg value="${language}/index.xml" />
+        <arg value="${dbk}/xhtml5/chunk.xsl" />
+        <arg value="use.id.as.filename=1" />
+        <arg value="html.ext=.html"/> 
+        <arg value="base.dir=${language}/xhtml5" />
```
[the remark this answers is not in the document]
Comment (kolekti) [reply doc_src/build.xml]: These are temporary directories. These should be deleted in the doc build process.

[the remark this answers is not in the document]
Comment (amake) [reply doc_src/build.xml]: OK, thanks.

File: doc_src/build.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (amake) [reply doc_src/build.xml]: >  That's also why every OmegaT.css stylesheet shiould remain in doc_src/$language directories.

I just realized that the `css` target and the `instant-start` target both copy different files to `../docs/${language}/OmegaT.css`. That seems like it should be a problem.

Or does the CSS get embedded in the output?

**Comment by amake** [comment]
Here are the remaining issues I'm seeing:

- The Instant Start Guide and the HTML5 manual both live in the same output directory and both want to reference `OmegaT.css`, but each one is supposed to have different CSS content. This is untenable.
- I see that the HTML5 pages generated in `doc/*/*.html` reference a `docbook.css`, which is present in `doc_src/*/xhtml5` but is never copied to `doc/*/`. Is this needed or not?
- The generated HTML5 pages load JQuery from `ajax.googleapis.com` rather than a bundled copy. Not only that, but when it *can't* load JQuery (no network connection, etc.) the page doesn't gracefully degrade, but rather seems not to load at all.

**Comment by miurahr** [comment]
Generated html5 document refers 'docbook.css' but it seems not to exist.
```
      <link href="docbook.css" rel="stylesheet" type="text/css" />
      <link href="OmegaT_webhelp.css" rel="stylesheet" type="text/css" />
```

**Comment by amake** [comment]
@kolekti Thanks for the updates on SourceForge. I've applied them here.

Can you comment as to what should be done about `docbook.css`? If it isn't used, can the reference to it be suppressed?

I looked briefly at the generated docs and noticed this issue with the code blocks:

image

Changing this

```
pre {
    white-space: pre;
}
```

to this

```
pre {
    white-space: normal;
}
```

will fix it, but is there a better solution?

**Comment by amake** [comment]
Thank you for your help. I've taken the additional changes from [here](https://sourceforge.net/u/waloo/omegat/ci/58d56e16804a4e8e68eaeef7f3742803340573bc/log/?path=) and committed to `master` on SourceForge.
